code/modules/spells/spell_types/undirected/touch/_touch.dm

```diff
@@ -121,7 +121,7 @@
 /datum/action/cooldown/spell/undirected/touch/proc/register_hand_signals()
 	SHOULD_CALL_PARENT(TRUE)
 
-	RegisterSignal(attached_hand, COMSIG_ITEM_AFTERATTACK, PROC_REF(on_hand_hit))
+	RegisterSignal(attached_hand, list(COMSIG_ITEM_AFTERATTACK, COMSIG_ITEM_AFTERATTACK_SECONDARY), PROC_REF(on_hand_hit))
```

Comment:
you don't need to do this because by default it does regular attackby on right click, I have already tested this works as it was